packages/types/src/codec/BTreeMap.spec.ts

```diff
@@ -0,0 +1,76 @@
+import '../injector';
+import BTreeMap from './BTreeMap';
+import Text from '../primitive/Text';
+import U32 from '../primitive/U32';
+import { CodecTo } from '../types';
+import Struct from './Struct';
+
+const expectedMap = new Map<Text, U32>();
+expectedMap.set(new Text('bazzing'), new U32(69));
+
+describe('BTreeMap', (): void => {
+  describe('decoding', (): void => {
+    const testDecode = (type: string, input: any): void =>
+      it(`can decode from ${type}`, (): void => {
+        const s = new BTreeMap(Text, U32, input);
+
+        expect(s.toString()).toBe('{"bazzing":69}');
+      });
+
+    testDecode('map', expectedMap);
+    testDecode('hex', '0x041c62617a7a696e6745000000');
+    testDecode('Uint8Array', Uint8Array.from([4, 28, 98, 97, 122, 122, 105, 110, 103, 69, 0, 0, 0]));
+  });
+
+  describe('encoding', (): void => {
+    const testEncode = (to: CodecTo, expected: any): void =>
+      it(`can encode ${to}`, (): void => {
+        const s = new BTreeMap(Text, U32, expectedMap);
+        expect(s[to]()).toEqual(expected);
+      });
+
+    testEncode('toHex', '0x041c62617a7a696e6745000000');
+    testEncode('toJSON', { bazzing: 69 });
+    testEncode('toU8a', Uint8Array.from([4, 28, 98, 97, 122, 122, 105, 110, 103, 69, 0, 0, 0]));
```

Comment:
I think we need one test to just reflect for than one entry as well, so make sure the . in/out is fine. (Possibly "steal" one from the scale-codec, so at least we know we have parity on both sides)

Reply:
I took the fixture from https://github.com/paritytech/parity-scale-codec/blob/4a559de5f6fece90205a159e008b86304d4c8750/src/encode_like.rs#L130. It seems arbitrary, but at least now the JavaScript and the Rust tests use the same fixtures.